File: 프로그래머스/2/12973. 짝지어 제거하기/짝지어 제거하기.py
def solution(s):

    stack = []  # list가 파이썬에서 스택으로 가장 빠름
    for ch in s:
        if stack and stack[-1] == ch:
            stack.pop()
        else:
            stack.append(ch)
    return 1 if not stack else 0

File: 프로그래머스/2/12973. 짝지어 제거하기/짝지어 제거하기.java
import java.util.Stack;
class Solution
{
    public int solution(String s)
    {
        int answer = 0;

        Stack<Character> stack = new Stack();

        for(char c : s.toCharArray()) {
            if(stack.size() == 0) {
                stack.push(c);
            }
            else {
                char temp = stack.pop();
                // 짝이 아닐 때
                if(temp != c) {
                    stack.push(temp);
                    stack.push(c);
                }
            }
        }

        if(stack.size() == 0) {
            answer = 1;
        }
        
        return answer;
    }
}